spec/requests/api/editor/editor_can_put_articles_to_archive_spec.rb

```diff
@@ -1,37 +1,41 @@
-RSpec.describe 'DELETE /api/articles/:id' do
+RSpec.describe 'PUT /api/articles/:id' do
   let(:article) { create(:article) }
   let(:editor) { create(:user, role: 'editor') }
   let(:editor_headers) { editor.create_new_auth_token }
 
   describe 'Successfully' do
     before do
-      delete "/api/articles/#{article.id}", headers: editor_headers
+      put "/api/articles/#{article.id}",
+          params: { status: 'archived' },
+          headers: editor_headers
     end
 
     it 'is expected to return status code 200' do
       expect(response).to have_http_status 200
     end
 
     it 'is expected to return a success message' do
-      expect(response_json['message']).to eq 'The article was successfully deleted'
+      expect(response_json['message']).to eq 'Your article has been successfully updated!'
     end
 
-    it 'is expected to have deleted article' do
-      expect(Article.all.count).to eq 0
+    it 'is expected to have archive article' do
+      expect(Article.where(status: 'archived').count).to eq 1
```

Comment:
I haven't really had the chance yet to work with enum's so correct me if i'm wrong.
cant we test this `expect(Article.where(status: 'archived').count).to eq 1`
in
`editor_can_publish_articles_spec.rb` 
like 
`expect(Article.where(status: 'published').count).to eq 10` ?

Reply:
yes we can, only it will be 1 article. enums also give possibility to check status as .published?, it will give true in this case.